src/shipit_static_analysis/tests/test_revisions.py

```diff
@@ -33,14 +34,30 @@ def test_phabricator(mock_phabricator):
     })
 
     r = PhabricatorRevision('51:PHID-DIFF-testABcd12', api)
-    assert r.mercurial == 'coffeedeadbeef123456789'
+    assert not hasattr(r, 'mercurial')
     assert r.diff_id == 42
     assert r.diff_phid == 'PHID-DIFF-testABcd12'
     assert r.url == 'https://phabricator.test/PHID-DIFF-testABcd12/'
     assert r.build_diff_name() == 'PHID-DIFF-testABcd12-clang-format.diff'
     assert r.id == 51  # revision
     assert r.phid == 'PHID-DREV-zzzzz'
 
+    # Load full patch
+    assert r.patch is None
+    r.apply(mock_repository)
+    assert r.patch is not None
+    assert isinstance(r.patch, str)
+    assert len(r.patch.split('\n')) == 7
+    patch = Patch.parse_patch(r.patch)
+    assert patch == {
+        'test.txt': {
+            'touched': [],
+            'deleted': [],
+            'added': [2],
+            'new': False
+        }
+    }
+
```

Comment:
maybe also assert that the file exists?

Reply:
to make sure the patch was actually applied

Reply:
The patch does not create the file, but i can check its content.